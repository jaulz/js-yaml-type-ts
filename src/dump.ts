export default function dump(value?: any): string {
  if (typeof value === 'undefined') {
    return 'undefined'
  }

  if (value === null) {
    return 'null'
  }

  if (typeof value === 'string') {
    return `'${value}'`
  }

  if (typeof value === 'function') {
    return value.toString()
  }

  if (typeof value === 'number') {
    return `${value}`
  }

  if (Array.isArray(value)) {
    const inner = value.map(dump).join(', ')
    return `[ ${inner} ]`
  }

  if (typeof value === 'object') {
    const inner = Object.entries(value)
      .map(([key, val]) => `${key}: ${dump(val)}`)
      .join(', ')
    return `{ ${inner} }`
  }

  return JSON.stringify(value)
}
