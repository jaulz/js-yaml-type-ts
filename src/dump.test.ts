import dump from './dump'

describe('dump', () => {
  it('is a function', () => {
    expect(dump).toBeDefined()
  })

  it('dumps undefined', () => {
    expect(dump()).toMatchSnapshot()
    expect(dump(undefined)).toMatchSnapshot()
  })

  it('dumps null', () => {
    expect(dump(null)).toMatchSnapshot()
  })

  it('dumps string', () => {
    expect(dump('test')).toMatchSnapshot()
  })

  it('dumps number', () => {
    expect(dump(3)).toMatchSnapshot()
  })

  it('dumps function', () => {
    expect(dump(() => true)).toMatchSnapshot()
  })

  it('dumps class', () => {
    expect(
      dump(
        class Test {
          test() {
            return true
          }
        }
      )
    ).toMatchSnapshot()
  })

  it('dumps flat object', () => {
    expect(
      dump({
        1: 2,
        '2': 'string',
        3: false,
      })
    ).toMatchSnapshot()
  })

  it('dumps deep object', () => {
    expect(
      dump({
        1: 2,
        '2': 'string',
        3: [
          'array',
          {
            undefined: undefined,
            null: null,
            custom: 'object',
            with: 'multiple',
            keys: (and: any) => {
              return 'functions'
            },
          },
        ],
      })
    ).toMatchSnapshot()
  })
})
