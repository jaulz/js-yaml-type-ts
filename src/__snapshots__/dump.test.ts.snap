// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dump dumps class 1`] = `
"function Test() {
            }"
`;

exports[`dump dumps deep object 1`] = `
"{ 1: 2, 2: 'string', 3: [ 'array', { undefined: undefined, null: null, custom: 'object', with: 'multiple', keys: function (and) {
                        return 'functions';
                    } } ] }"
`;

exports[`dump dumps flat object 1`] = `"{ 1: 2, 2: 'string', 3: false }"`;

exports[`dump dumps function 1`] = `"function () { return true; }"`;

exports[`dump dumps null 1`] = `"null"`;

exports[`dump dumps number 1`] = `"3"`;

exports[`dump dumps string 1`] = `"'test'"`;

exports[`dump dumps undefined 1`] = `"undefined"`;

exports[`dump dumps undefined 2`] = `"undefined"`;
